File: daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        vector<int> sols(temp.size(), 0);
        stack<int> t; 
        for (int i = 0; i < temp.size(); i++){
            while (!t.empty() && temp[i] > temp[t.top()]){
                sols[t.top()] = i - t.top(); 
                t.pop();
            }
            t.push(i);
        }
        return sols;
    }
};

File: daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] t) {

        Stack<Integer> stack = new Stack<>();
        int[] sols = new int[t.length];
        for (int i = 0; i < t.length; i++){
            while (!stack.isEmpty() && t[stack.peek()] < t[i]){
                sols[stack.peek()] = i - stack.peek();
                stack.pop();
            } 
            stack.push(i);
        }

        while (!stack.isEmpty()){
            sols[stack.peek()] = 0; 
            stack.pop();
        }
        return sols;
    }
}